src/bin/common/solving_loop.ml

```diff
@@ -410,6 +417,44 @@ let main () =
       Loc.report loc name ty DStd.Term.print value
   in
 
+  let set_sat_solver sat st =
+    let optim = State.get optimize_key st in
+    match sat with
+    | Util.Tableaux | Tableaux_CDCL when optim ->
+      warning
+        "Sat-solver %a is incompatible with optimization: ignoring command."
+        Util.pp_sat_solver
+        sat;
+      st
+    | Tableaux | Tableaux_CDCL | CDCL | CDCL_Tableaux ->
+      O.set_sat_solver sat;
+      (* `make_sat` returns the sat solver corresponding to the new sat_solver option. *)
+      State.set
+        sat_solver_key
+        (make_sat ())
+        st
+  in
+
+  let set_optimize optim st =
+    let () =
+      if optim then
+        DStd.Extensions.Smtlib2.(enable maxsmt)
+      else
+        DStd.Extensions.Smtlib2.(disable maxsmt)
+    in
+    let sat, _ = State.get sat_solver_key st in
+    match sat with
+    | Util.Tableaux | Tableaux_CDCL when optim ->
+      warning
+        "Sat-solver %a is incompatible with optimization: ignoring command."
+        Util.pp_sat_solver
+        sat;
```

Comment:
You are not ignoring the command here since you call `DStd.Extensions.Smtlib2.(enable maxsmt)` above.